### PR TITLE
4MB report size limit for A2-installed Chef Infra Server

### PR DESCRIPTION
**Pull request by sean-horn**
Signed-off-by: Sean Horn <sean_horn@chef.io>

### :nut_and_bolt: Description: What code changed, and why?

Update the standard installs that can be changed to reflect the 4MB GRPC limit.
This max report size applies to the allowable chef-client and Compliance profile reports proxied through a Chef Infra Server.

Proactively prevent 413/429 problems during report sending.

### :white_check_mark: Checklist

- [X] I have read the [CONTRIBUTING document](https://github.com/chef/automate/blob/master/CONTRIBUTING.md).
- [X] Docs added/updated?
- [X] All commits have been signed-off for the [Developer Certification of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

### CODE REVIEW

**Comment by yzl** [comment]
@chef/automate-deployment-team Is there a reason not to change the default value for the Infra Server's report size?

**Comment by ryancragun** [comment]
@sean-horn @yzl We recently changed the default value to 2MB, which is the current default of the Chef Infra Server. I think I'd rather just increase the default than add required config changes https://github.com/chef/automate/blob/master/api/config/erchef/config_request.go#L46

File: components/automate-chef-io/content/docs/infra-server.md [hunk 1]
```diff
@@ -61,6 +61,19 @@ Install Chef Automate and Chef Infra Server on the same host with this command:
 sudo chef-automate deploy --product automate --product infra-server
 ```
 
+Add a stanza to a new patch.toml that bumps the max report size
+
+```toml
+[erchef.v1.sys.api]
+max_request_size = 4000000
```
Comment (kagarmoe) [inline components/automate-chef-io/content/docs/infra-server.md:68]:
```suggestion
max_request_size = 2000000
```

Comment (kagarmoe) [inline components/automate-chef-io/content/docs/infra-server.md:68]:
```suggestion
max_request_size = 2000000
```

File: components/automate-chef-io/content/docs/infra-server.md [hunk 2]
```diff
@@ -74,11 +87,14 @@ Installations require elevated privileges, so run the commands as the superuser
...
 
-1. Add a stanza to the configuration file to deploy Chef Automate and Chef Infra Server:
+1. Add a stanza to the configuration file to deploy Chef Automate and Chef Infra Server and bump the max report size:
 
     ```toml
       [deployment.v1.svc]
       products=["automate", "infra-server"]
+
+      [erchef.v1.sys.api]
+      max_request_size = 4000000
```
Comment (kagarmoe) [inline components/automate-chef-io/content/docs/infra-server.md:97]:
```suggestion
      max_request_size = 2000000      
```

File: components/automate-chef-io/content/docs/infra-server.md [hunk 3]
```diff
@@ -109,11 +125,14 @@ Installations require elevated privileges, so run the commands as the superuser
...
 
-1. Add a stanza to the configuration file to disable Chef Automate data collection:
+1. Add a stanza to the configuration file to disable Chef Automate data collection and bump the max report size:
 
     ```toml
        [erchef.v1.sys.data_collector]
        enabled = false
+
+      [erchef.v1.sys.api]
+      max_request_size = 4000000
```
Comment (kagarmoe) [inline components/automate-chef-io/content/docs/infra-server.md:135]:
```suggestion
      max_request_size = 2000000      
```

File: components/automate-chef-io/content/docs/infra-server.md [hunk 4]
```diff
@@ -146,6 +165,9 @@ Installations require elevated privileges, so run the commands as the superuser
        # Disable Automate data collection as Automate will not be deployed
        [erchef.v1.sys.data_collector]
        enabled = false
+
+      [erchef.v1.sys.api]
+      max_request_size = 4000000
```
Comment (kagarmoe) [inline components/automate-chef-io/content/docs/infra-server.md:170]:
```suggestion
      max_request_size = 2000000      
```

File: components/automate-chef-io/content/docs/infra-server.md [hunk 5]
```diff
@@ -160,17 +182,20 @@ Installations require elevated privileges, so run the commands as the superuser
...
 
-1. Create a `patch.toml` file to add `infra-server` to the list of products to deploy:
+1. Create a `patch.toml` file to add `infra-server` to the list of products to deploy and bump the max report size:
 
     ```toml
        [deployment.v1.svc]
        products=["automate", "infra-server"]
+
+      [erchef.v1.sys.api]
+      max_request_size = 4000000
```
Comment (kagarmoe) [inline components/automate-chef-io/content/docs/infra-server.md:192]:
```suggestion
      max_request_size = 2000000      
```

**Comment by ryancragun** [comment]
I believe I need to clarify my position as a suggestion and fact we're muddled together.
* We recently changed the default value in question to 2MB to be in line with the default in the Chef Infra Server omnibus package.
* If customers are running into issues with this value often enough that we deem necessary to document it, we should probably increase the default value instead instead of more documentation, as that's a more delightful customer outcome.

**Comment by kagarmoe** [comment]
@ryancragun 
1. Is the 2MB the correct default?
2. What is the maximum `max_request_size` value?

File: components/automate-chef-io/content/docs/infra-server.md (hunk 1 above)
Comment (kagarmoe) [inline components/automate-chef-io/content/docs/infra-server.md:68]:
```suggestion
max_request_size = 2000000
# Default: 2000000 (2mb), increase to 4000000 (4mb) as needed.
```

**Pull request merged**